File: Algorithms/ Sorting/Quicksort 1 - Partition.php
<?php

/*
 * Complete the 'quickSort' function below.
 *
 * The function is expected to return an INTEGER_ARRAY.
 * The function accepts INTEGER_ARRAY arr as parameter.
 */

function quickSort($arr) {
    # Write your code here
    if (count($arr) <= 1) {
        return $arr;
    }
    else {
        $pivot = $arr[0];
        $less = [];
        $greater = [];
        for ($i = 1; $i < count($arr); $i++) {
            if ($arr[$i] <= $pivot) {
                $less[] = $arr[$i];
            }
            else {
                $greater[] = $arr[$i];
            }
        }
        return array_merge(quickSort($less), [$pivot], quickSort($greater));
    }
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$n = intval(trim(fgets(STDIN)));

$arr_temp = rtrim(fgets(STDIN));

$arr = array_map('intval', preg_split('/ /', $arr_temp, -1, PREG_SPLIT_NO_EMPTY));

$result = quickSort($arr);

fwrite($fptr, implode(" ", $result) . "\n");

fclose($fptr);

File: Algorithms/Sorting/Quicksort 1 - Partition.cs
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'quickSort' function below.
     *
     * The function is expected to return an INTEGER_ARRAY.
     * The function accepts INTEGER_ARRAY arr as parameter.
     */

    public static List<int> quickSort(List<int> arr)
    {
        if (arr.Count <= 1) {
            return arr;
        }
        int pivotIndex = arr.Count / 2;
        int pivot = arr[pivotIndex];
        arr.RemoveAt(pivotIndex);
        List<int> left = new List<int>();
        List<int> right = new List<int>();
        foreach (int num in arr)
        {
            if (num <= pivot)
            {
                left.Add(num);
            }
            else
            {
                right.Add(num);
            }
        }
        List<int> sorted = new List<int>();
        sorted.AddRange(quickSort(left));
        sorted.Add(pivot);
        sorted.AddRange(quickSort(right));
        return sorted;
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<int> arr = Console.ReadLine().TrimEnd().Split(' ').Select(arrTemp => Convert.ToInt32(arrTemp)).ToList();

        List<int> result = Result.quickSort(arr);

        textWriter.WriteLine(String.Join(" ", result));

        textWriter.Flush();
        textWriter.Close();
    }
}

